src/Forms/Schema/FormSchema.php

```diff
@@ -112,9 +113,29 @@ public function getSchema(Form $form)
             $schema['fields'][] = $field->getSchemaData();
         }
 
+        // Validate there are react components for all fields
+        // Note 'actions' (FormActions) are always valid because FormAction.schemaComponent has a default value
+        $this->recursivelyValidateSchemaData($schema['fields']);
+
         return $schema;
     }
 
+    private function recursivelyValidateSchemaData(array $schemaData)
+    {
+        foreach ($schemaData as $data) {
+            if (!$data['schemaType'] && !$data['component']) {
+                $name = $data['name'];
+                $message = "Could not find a react component for $name."
+                    . " Set a schemaDataType or schemaComponent on the field class,"
+                    . " or replace or remove the field instance from the field list.";
```

Comment:
`schemaDataType` and `schemaComponent` don't match with the names of the data in the condition... I recommend just removing this line and just say to remove/replace the field instance. Most of the time it'll be developers trying to add pre-existing fields, not making their own fields.
```suggestion
                    . " Replace or remove the field instance from the field list.";
```

If you really want to keep the guidance about adding a type or component to the field class, it needs to be a lot clearer what those actually are, since they don't match up with what's in the condition, and people tracing this back will come to this line in code since this is where the exception is being thrown from.

Reply:
Have updated the message and put replace/remove first.  I've reworded the 'update field class' bit and mentioned the word 'property'.  This message is also aimed at people in CMS squad doing product dev and the a lot of the time adding one of the two properties will be correct.